### ISSUE TITLE
[Proposal] "If Call" syntax sugar

### ISSUE THREAD

**Issue by xieguigang** [dotnet/vblang#32]
Hi,

Assuming that we have a single method call in a single ``If`` test block, and then ``If..Then`` will be used:

```vbnet
Dim test As Boolean

' If test Then Call method()
' Or
If test Then
    Call method()
End If

Dim i%

If i = 10 Then
    Call method()
End If

Dim test2 = Function(str$) As Boolean
                ' blabla
            End Function

If test2(str:="blabla") Then
    Call method()
End If

' If want call multiple lines sub program
If Not test2(str:="blabla") Then
    Call Sub()
             ' blabla
         End Sub
End If
```

And I think this situation can be simplify by using a ``If Call`` syntax sugar

```vbnet
Dim test As Boolean

Call test ? method()

Dim i%

Call (i = 10) ? method()

Dim test2 = Function(str$) As Boolean
                ' blabla
            End Function

Call test2(str:="blablabla") ? method()

' If want call multiple lines sub program
Call (Not test2(str:="blabla")) ? Sub()
                                      ' balbala
                                  End Sub
```

**Comment by rskar-git** [comment]
I believe the single line If statement already comes close enough:
```
    If test Then method()
    If i = 10 Then method()
    If Not test2(str:="blabla") Then Call (Sub()
        'balbala
        End Sub)()
```
The above are all valid syntax today! See https://msdn.microsoft.com/en-us/library/752y8abs.aspx

**Comment by Bill-McC** [comment]
Don't like. If I see 
`Call testsomereallylongmetodname(withparamater1, andparametertwo, andmaybeafunctioncalltogetanotherparameter, andetc)`
I shouldn't have to scan to the end/middle to see if there is a ? which might change the meaning completely. Traditionally, Call SomeFunction means to ignore the return value, but this syntax is suggesting a boolean test on the return value.
So don't like, and don't see any real need.

**Comment by nathanstites** [comment]
Isn't this exactly what the classic On keyword did?

**Comment by rskar-git** [comment]
@nathanstites No, not all. The On keyword of On ... GoTo and On ... GoSub was about making a jump-table - see https://msdn.microsoft.com/en-us/library/aa266175(v=vs.60).aspx.

**Comment by nathanstites** [comment]
@rskar-git You are correct.  My memory appears to have corrupted the concept here.  
For some reason I thought it was simply: On A B
Where if A was something truthy B would occur which was typically a GOSUB or GOTO.

At any rate thanks for the link.  
Using ON for this rather than a ternary CALL statement seems more BASICy to me.  Apparently so much so I incorrectly recalled it being that way.
